docs/concepts/how-ipfs-works.md

```diff
@@ -5,86 +5,171 @@ description: Learn how the InterPlanetary File System (IPFS) works and why it's
+Because IPFS is an open-source protocol with multiple implementations, an IPFS node is not required to implement full routing functionality itself. Instead, an IPFS node can request that a <VueCustomTooltip label="An IPFS node that perform tasks on behalf of network peers using the IPFS HTTP API." underlined multiline is-medium>delegated router</VueCustomTooltip> perform a routing task for it. For example, if an IPFS node does not implement the DHT, a delegated router can search the DHT for peers on its behalf. The main benefit of delegated routing is that nodes are not required to implement routing functionality themselves if they do not wish to, or do not have the computing resources to do so.
 
-So, to recap, IPFS lets you give CIDs to content and link that content together in a Merkle DAG. Now let's move on to the last piece: how you find and move content.
+## How IPFS transfers data
 
-## Distributed hash tables (DHTs)
+In addition to [routing data](#how-ipfs-routes-data), nodes in the IPFS network must efficiently distribute and deliver the content addressed data, taking into account that there are some nodes in the network who already have a copy of the data, and other nodes who do not have a copy of the data, but want one. To handle the transfer of data, IPFS uses the following subsystems:
 
-To find which peers are hosting the content you're after (_discovery_), IPFS uses a [distributed hash table](dht.md), or DHT. A hash table is a database of keys to values. A _distributed_ hash table is one where the table is split across all the peers in a distributed network. To find content, you ask these peers.
+- [IPFS HTTP Gateways](#ipfs-http-gateways)
+- [GraphSync](#graphsync)
+- [Sneakernet](#sneakernet)
 
-The [libp2p project](https://libp2p.io/) is the part of the IPFS ecosystem that provides the DHT and handles peers connecting and talking to each other. (Note that, as with IPLD, libp2p can also be used as a tool for other distributed systems, not just IPFS.)
+### IPFS HTTP Gateways
 
-Once you know where your content is (or, more precisely, which peers are storing each of the blocks that make up the content you're after), you use the DHT again to find the current location of those peers (_routing_). So, to get to the content, use libp2p to query the DHT twice.
+HTTP Gateways allow applications that do not support or implement the IPFS protocol to interact with nodes on the IPFS network. There are multiple types of HTTP gateways, each optimized for specific tasks, and other factors like security and performance. Examples include:
```

Comment:
This categorisation of gateways isn't useful, e.g. writeable gateways aren't a thing and ones that require auth aren't key to understanding how IPFS works.


```suggestion
HTTP Gateways allow applications that do not support or implement all IPFS subsystems to fetch data from the IPFS network using an HTTP interface. In its simplest form, a gateway is an IPFS Node that also exposes an [HTTP Gateway API](https://github.com/ipfs/specs/blob/main/http-gateways/README.md)"
```